app/src/main/java/com/banuba/quickstart_c_api/Image.java

```diff
@@ -0,0 +1,37 @@
+package com.banuba.quickstart_c_api;
+
+import java.nio.ByteBuffer;
+
+public class Image {
+
+    Image() {}
+
+    Image(byte[] image0, byte[] image1, byte[] image2, int w, int h) {
+        if(image0 != null) {
+            mImageZero =  ByteBuffer.wrap(image0);
+        }
+        if(image1 != null) {
+            mImageFirst =  ByteBuffer.wrap(image1);
+        }
+        if(image2 != null) {
+            mImageSecond =  ByteBuffer.wrap(image2);
+        }
+
+        mWidth = w;
+        mHeight = h;
+    }
+
+    public ByteBuffer mImageZero;
```

Comment:
final private ByteBuffer mPlane0;     etc.

Reply:
I made this field private but not final because their values have changed in the MainActivity

Reply:
These values should not change, only in constructor. They must be final.